gramex/config.py

```diff
@@ -840,3 +844,78 @@ def used_kwargs(method, kwargs, ignore_keywords=False):
             target = used if key in set(argspec.args) else rest
             target[key] = val
     return used, rest
+
+
+esl_paths = AttrDict()
+esl_paths['source'] = Path(__file__).absolute().parent
+esl_conf = (+PathConfig(esl_paths['source'] / 'gramex.yaml')).get('eslog', AttrDict())
+# The above config needs to be fixed. Gramex is loading this as a service
+allowed_log_levels = ['INFO', 'ERROR', 'DEBUG', 'WARN']
+es_q = queue.Queue()
+
+
+class ConnSingleton:
+    def __init__(self, cls):
+        self.cls = cls
+        self.instance = None
+
+    def __call__(self, *args, **kwargs):
+        if self.instance == None:
+            self.instance = self.cls(*args, **kwargs)
+        return self.instance
+
+
+@ConnSingleton
+class ElDb:
+    '''
+    Maintains ElasticSearch connection. Picks the ES configuration from 
+    environment variables.
+    '''
+    connection = None
+    def get_connection(self):
+        # check if connection is created or still alive
+        if self.connection is None or not self.connection.ping():
+            self.connection = Elasticsearch(variables['ES_HOST'],
+                http_auth=(variables['ES_USER'], variables['ES_PASS']))
+        return self.connection
```

Comment:
Can we remove all classes here? We're not really re-using them. Instead a cached function like this could work:

```
es_conf = {}
def get_connection():
    if 'connection' not in es_conf:
        es_conf['connection'] = ...
     return es_conf['connection']
```